src/ophys_etl/modules/segmentation/seed/seeder.py

```diff
@@ -0,0 +1,305 @@
+
+    """
+    def __init__(self,
+                 keep_fraction: float = 0.4,
+                 seeder_grid_size: int = 5,
+                 *args, **kwargs):
+        self._seeder_grid_size = seeder_grid_size
+        self._keep_fraction = keep_fraction
+        self._seed_image: np.ndarray = None
+        super(ImageBlockMetricSeeder, self).__init__(*args, **kwargs)
+
+    def select_seeds(self, image: np.ndarray, sigma: Optional[float] = None):
+        """select seeds from an image
+
+        Parameters
+        ----------
+        image: np.ndarray
+            an image whose intensity values are the metrics for determining
+            seeds
+        sigma: float
+            if provided, will be the Gaussian filter sigma for a smoothing
+            operation on the image before any additional processing.
+
+        """
+        self._fov_shape = image.shape
+
+        if sigma is not None:
+            image = gaussian_filter(image, mode="constant", sigma=sigma)
+        self._seed_image = image
+
+        row = 0
+        seeds = []
+        while row < image.shape[0]:
+            col = 0
+            while col < image.shape[1]:
+                block = image[row: (row + self._seeder_grid_size),
+                              col: (col + self._seeder_grid_size)]
+                inds = np.array(
+                        np.unravel_index(np.argmax(block), block.shape))
+                inds += np.array([row, col])
+                seeds.append(
+                        Seed(coordinates=tuple(inds),
+                             value=block.max(),
+                             excluded_by_roi=False,
+                             excluded_by_quantile=False))
+                col += self._seeder_grid_size
+            row += self._seeder_grid_size
+
+        seeds_sorted = sorted(seeds,
+                              key=lambda x: x['value'],
+                              reverse=True)
+        num_keep = int(self._keep_fraction * len(seeds_sorted))
+        self._candidate_seeds = seeds_sorted[:num_keep]
+        self._excluded_seeds = seeds_sorted[num_keep:]
+        for i in range(len(self._excluded_seeds)):
+            self._excluded_seeds[i]["exclusion_reason"] = \
+                    "excluded by fraction"
+
+    def log_to_h5_group(self,
+                        h5file: h5py.File,
+                        group_name: str = "seeding"):
+        """records some attributes of this seeder to an hdf5 group for
+        inspection purposes.
+
+        Parameters
+        ----------
+        h5file: h5py.File
+            an h5py.File object which has been opened in a writable mode
+        group_name: str
+            the name of the group to which this object's logged attrbiutes
+            will be inserted
+
+        """
+        # collect any candidates not even considered (perhaps a segmenter
+        # can impose a stopping condition different from running out of seeds
+        for seed in self._candidate_seeds:
```

Comment:
I got the impression from the class docstring that this method would be called once before and once after segmentation. Should we avoid altering `self._candidate_seeds` and `self._excluded_seeds` in this logging method?

Reply:
I think the docstring part you are talking about is:
```
segmenter.log_to_h5_group(f, "segmenting")
...
seeder.log_to_h5_group(f, "seeding")
```
where I am anticipating that future refined segmentation implementations also log something to an hdf5 group, and the seeder and segmentation groups share the same hdf5 file.

Reply:
so, I expect to log the seeder once at the end

Reply:
Got it. I definitely only read the first two letters of `segmenter.`